samples/deploy/crd/v1beta1/ApisixRoute.yaml

```diff
@@ -225,7 +225,6 @@ spec:
                             type: string
                     required:
                       - enable
-                      - type
```

Comment:
`type` should be a required field. Why change this?

Reply:
user can only setting 
```yaml
   authentication:     
       enable: false
```

if set type required, they must set one of `keyAuth` or `basicAuth` type.

Reply:
But seems like we don't have a default authentication way. When people enable auth but don't configure the specific authentication, actually no authentication is enabled, this behavior violated the semantic of field `enable`.

Reply:
let me set the default value to ｋｅｙＡｕｔｈ

Reply:
Got it.